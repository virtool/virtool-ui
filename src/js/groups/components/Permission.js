import React from "react";
import styled from "styled-components";
import { Checkbox, SelectBoxGroupSection } from "../../base";

const StyledGroupPermission = styled(SelectBoxGroupSection)`
    user-select: none;
`;

export const GroupPermission = ({ active, permission, onClick }) => (
    <StyledGroupPermission active={active} onClick={onClick}>
        <Checkbox checked={active} label={permission} />
    </StyledGroupPermission>
);
