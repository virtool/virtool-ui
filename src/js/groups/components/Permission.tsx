import { Checkbox, SelectBoxGroupSection } from "@base";
import React from "react";
import styled from "styled-components";

const StyledGroupPermission = styled(SelectBoxGroupSection)`
    background-color: ${props => (props.active ? props.theme.color.blue : "white")};
    user-select: none;
`;

type GroupPermissionProps = {
    /** Whether the permission is currently active */
    active: boolean;
    /** The name of the permission */
    permission: string;
    /** Handles click event when permission is clicked */
    onClick: () => void;
};

/**
 * Group permission checkbox
 */
export function GroupPermission({ active, permission, onClick }: GroupPermissionProps) {
    return (
        <StyledGroupPermission active={active} onClick={onClick}>
            <Checkbox checked={active} label={permission} />
        </StyledGroupPermission>
    );
}
