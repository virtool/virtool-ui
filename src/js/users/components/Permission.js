import React from "react";
import styled from "styled-components";
import { BoxGroupSection, Icon } from "../../base";

const StyledPermissionIcon = styled(Icon)`
    margin-right: 15px;
`;

const PermissionIcon = ({ value }) => (
    <StyledPermissionIcon name={value ? "check" : "times"} color={value ? "green" : "red"} fixedWidth />
);

const StyledPermissionItem = styled(BoxGroupSection)`
    align-items: center;
    display: flex;
`;

export const PermissionItem = ({ permission, value }) => {
    if (value) {
        return (
            <StyledPermissionItem>
                <PermissionIcon value={value} /> <code>{permission}</code>
            </StyledPermissionItem>
        );
    }
    return (
        <StyledPermissionItem>
            <PermissionIcon value={value} /> <code>{permission}</code>
        </StyledPermissionItem>
    );
};
