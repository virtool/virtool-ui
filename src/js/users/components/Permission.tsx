import React from "react";
import styled from "styled-components";
import { BoxGroupSection, Icon } from "../../base";

const StyledPermissionIcon = styled(Icon)`
    margin-right: 15px;
`;

const StyledPermissionItem = styled(BoxGroupSection)`
    align-items: center;
    display: flex;
`;

type PermissionItemProps = {
    /* Name of the permission */
    permission: string;
    /* Indicates if the permission is granted */
    value: boolean;
};

/**
 * A condensed permission item for use in a list of user permissions
 */
export function PermissionItem({ permission, value }: PermissionItemProps) {
    return (
        <StyledPermissionItem aria-label={`${permission}:${value}`}>
            <StyledPermissionIcon name={value ? "check" : "times"} color={value ? "green" : "red"} fixedWidth />
            <code>{permission}</code>
        </StyledPermissionItem>
    );
}
