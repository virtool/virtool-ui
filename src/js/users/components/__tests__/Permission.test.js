import { PermissionItem } from "../Permission";

describe("<PermissionItem />", () => {
    let props;
    beforeEach(() => {
        props = {
            acc: [],
            value: "foo",
            permission: { foo: "bar" }
        };
    });

    it("should render", () => {
        const wrapper = shallow(<PermissionItem {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
