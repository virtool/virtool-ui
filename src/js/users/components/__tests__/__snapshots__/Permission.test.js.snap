// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PermissionItem /> should render 1`] = `
<Permission__StyledPermissionItem>
  <PermissionIcon
    value="foo"
  />
   
  <code>
    <Component />
  </code>
</Permission__StyledPermissionItem>
`;
